extra space in INI markers section causes inconsistent behavior

In the markers section of the INI file, if you include an extra space between the marker and the colon, the pytest --markers command lists the marker, however the running pytest -m mymarker causes an "Attribute Error: 'mymarker' not a registered marker.

pytest.ini example:

```
[pytest] 
addopts = -rsxX -l --strict -v -x

markers = 
	incomplete: tests which are not complete
	dev : tests in development
	complete: tests which are complete
	noDbUpdate: tests which do not update the database

filterwarnings =
    ignore

```

**NOTE: There's an extra space between dev and the colon.**

pytest --markers lists:

```
> pytest --markers
@pytest.mark.incomplete: tests which are not complete

@pytest.mark.dev : tests in development

@pytest.mark.complete: tests which are complete

@pytest.mark.noDbUpdate: tests which do not update the database

```

pytest -m dev generates:

```
>pytest -m dev
============================= test session starts =============================
platform win32 -- Python 3.4.3, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- c:\python34\python.exe
cachedir: .cache
rootdir: C:\Users\Greg\...\tests, inifile: pytest.ini
collecting 0 items / 1 errors
=================================== ERRORS ====================================
___________________ ERROR collecting test_createShipment.py ___________________
test_createShipment.py:127: in <module>
    @pytest.mark.dev
c:\python34\lib\site-packages\_pytest\mark.py:264: in __getattr__
    self._check(name)
c:\python34\lib\site-packages\_pytest\mark.py:279: in _check
    raise AttributeError("%r not a registered marker" % (name,))
E   AttributeError: 'dev' not a registered marker
!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!
=========================== 1 error in 1.06 seconds ===========================

```

Version info:
Python 3.4.3  running on Windows 10.0.14393
pytest version 3.2.3

--- a/_pytest/mark.py
+++ b/_pytest/mark.py
@@ -272,8 +272,9 @@ class MarkGenerator:
             pass
         self._markers = l = set()
         for line in self._config.getini("markers"):
-            beginning = line.split(":", 1)
-            x = beginning[0].split("(", 1)[0]
+            marker, _ = line.split(":", 1)
+            marker = marker.rstrip()
+            x = marker.split("(", 1)[0]
             l.add(x)
         if name not in self._markers:
             raise AttributeError("%r not a registered marker" % (name,))

--- a/changelog/2856.bugfix
+++ b/changelog/2856.bugfix
@@ -1,0 +1,1 @@
+Strip whitespace from marker names when reading them from INI config.

--- a/testing/test_mark.py
+++ b/testing/test_mark.py
@@ -169,6 +169,23 @@ def test_markers_option(testdir):
     ])
 
 
+def test_ini_markers_whitespace(testdir):
+    testdir.makeini("""
+        [pytest]
+        markers =
+            a1 : this is a whitespace marker
+    """)
+    testdir.makepyfile("""
+        import pytest
+
+        @pytest.mark.a1
+        def test_markers():
+            assert True
+    """)
+    rec = testdir.inline_run("--strict", "-m", "a1")
+    rec.assertoutcome(passed=1)
+
+
 def test_markers_option_with_plugin_in_current_dir(testdir):
     testdir.makeconftest('pytest_plugins = "flip_flop"')
     testdir.makepyfile(flip_flop="""\

